tests/time_tests/c++/main.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cli.h"
+#include "statistics_writer.h"
+#include "pipeline.h"
+
+#include <iostream>
+
+
+bool parseAndCheckCommandLine(int argc, char **argv) {
+    // ---------------------------Parsing and validating input arguments--------------------------------------
+    std::cout << "Parsing input parameters" << std::endl;
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_help || FLAGS_h) {
+        showUsage();
+        //showAvailableDevices();
+        return false;
+    }
+
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Model is required but not set. Please set -m option.");
+    }
+
+    return true;
+}
+
+
+int main(int argc, char **argv) {
+    if (!parseAndCheckCommandLine(argc, argv)) {
+        return -1;
+    }
+    runPipeline(FLAGS_m, FLAGS_d);
```

Comment:
Please check the return code. I also believe we should catch exceptions here. The goal is to ensure runPipeline does not abort main unexpectedly.

Reply:
Done